### PR TITLE
Remove include of "bionics.h" from "character.h".  

### PR DESCRIPTION
**Pull request by BevapDin**
Changes `Character::my_bionics` to be a `copyable_uniqe_ptr<bionic_collection>`.

`bionic_collection` is a simple wrapper for `std::vector<bionic>`, defined in "bionics.h".
It can be forward declared, without defining the `bionic` class (`std::vector<bionic>` can't be forward declared without defining the `bionic` class).

"bionics.h" used to be included (directly/indirectly) by 103 cpp files, no it's only included by 11 cpp files.

### CODE REVIEW

**Comment by kevingranade** [comment]
/usr/include/c++/7.2.0/bits/unique_ptr.h:322: typename std::add_lvalue_reference<_Tp>::type std::unique_ptr<_Tp, _Dp>::operator*() const [with _Tp = bionic_collection; _Dp = std::default_delete<bionic_collection>; typename std::add_lvalue_reference<_Tp>::type = bionic_collection&]: Assertion 'get() != pointer()' failed.
Aborted (core dumped)

backtrace from gdb:
Program terminated with signal SIGABRT, Aborted.
#0  0x00007feb97cd68a0 in raise () from /usr/lib/libc.so.6
(gdb) bt
#0  0x00007feb97cd68a0 in raise () from /usr/lib/libc.so.6
#1  0x00007feb97cd7f09 in abort () from /usr/lib/libc.so.6
#2  0x000056488d46f560 in std::__replacement_assert (__condition=0x56488d9231f7 "get() != pointer()", 
    __function=0x56488da63160 <_ZZNKSt10unique_ptrI17bionic_collectionSt14default_deleteIS0_EEdeEvE19__PRETTY_FUNCTION__> "typename std::add_lvalue_reference<_Tp>::type std::unique_ptr<_Tp, _Dp>::operator*() const [with _Tp = bionic_collection; _Dp = std::default_delete<bionic_collection>; typename std::add_lvalue_referen"..., __line=322, __file=0x56488d923008 "/usr/include/c++/7.2.0/bits/unique_ptr.h") at /usr/include/c++/7.2.0/x86_64-pc-linux-gnu/bits/c++config.h:472
#3  std::unique_ptr<bionic_collection, std::default_delete<bionic_collection> >::operator* (this=0x56488f540688) at /usr/include/c++/7.2.0/bits/unique_ptr.h:322
#4  Character::has_active_bionic (this=this@entry=0x56488f53fcc0, b=...) at src/character.cpp:630
#5  0x000056488d476f01 in Character::is_blind (this=this@entry=0x56488f53fcc0) at src/character.cpp:2212
#6  0x000056488d476fa1 in Character::recalc_sight_limits (this=this@entry=0x56488f53fcc0) at src/character.cpp:503
#7  0x000056488d433b37 in player::player (this=0x56488f53fcc0) at src/player.cpp:627
#8  0x000056488ca4bc28 in game::game (this=0x56488f4e5ae0) at src/game.cpp:242
#9  0x000056488c95af0f in init_global_game_state (mods=std::__debug::vector of length 1, capacity 1 = {...}) at test_main.cpp:77
#10 0x000056488c95bfd0 in main (argc=<optimized out>, argv=<optimized out>) at test_main.cpp:150

**Comment by kevingranade** [comment]
(gdb) f 4
#4  Character::has_active_bionic (this=this@entry=0x56488f53fcc0, b=...) at src/character.cpp:630
630         for (auto &i : *my_bionics) {
(gdb) p my_bionics
$1 = {<std::unique_ptr<bionic_collection, std::default_delete<bionic_collection> >> = std::unique_ptr<bionic_collection> containing 0x0, <No data fields>}

So... the unique ptr isn't initialized.

File: src/character.h [hunk 1]
```diff
@@ -602,7 +605,7 @@ class Character : public Creature, public visitable<Character>
         item weapon;
         item ret_null; // Null item, sometimes returns by weapon() etc
 
-        std::vector<bionic> my_bionics;
+        copyable_unique_ptr<bionic_collection> my_bionics;
```
Comment (kevingranade) [inline src/character.h]: This here is the problem I think, the vector was default contructed, but the unique_ptr is constructed empty, it just needs a new default constructed bionic_collection.

Comment (kevingranade) [reply src/character.h]: all good, added ```my_bionics.reset( new bionic_collection() );``` to the end of the Character constructor.

Comment (BevapDin) [reply src/character.h]: Yes, sorry. I forgot that. Somehow I assumed the wrapper would create the pointer automatically. May need to add something like this.

**Pull request merged**